### PR TITLE
Allow a request MSAL Redirect URI to pass broker validation.

### PR DESCRIPTION
**Pull request by rpdome**
Allow a request MSAL Redirect URI to pass broker validation.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java [hunk 1]
```diff
@@ -222,4 +222,14 @@ public static String getBrokerRedirectUri(final Context context, final String pa
...
                 signatureDigest);
     }
+
+    /**
+     * Helper method to get MSAL Redirect Uri
+     *
+     * @param clientId
+     * @return String : MSAL Redirect Uri
+     */
+    public static String getMsalRedirectUri(@NonNull final String clientId){
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: I was under the impression that MSAL with V2 broker's redirect URI should always be of the form msauth://<package_name>/<sign_hash>. Did the requirement change ?

Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: I'm not certain about this, but this redirect URL pattern is available as an option in the portal. We're also using it in the test app (which is why I was hitting this error).

Might be worth bringing this up during the standup.

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: The scheme suggested in the _new portal_ should now match the broker redirect uri

```
msauth://<package_name>/<signature_hash_decoded>
```

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: @danieldobalian can link to new docs

Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: Does this mean we're not supporting "msal" + clientId + "://auth"; anymore?

If yes, then I'll drop this PR.

Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java]: Yes that redirect is only supported for just standalone MSAL, not with broker.

**Pull request closed without merging**